shell/common/animator.cc

```diff
@@ -112,6 +116,33 @@ void Animator::BeginFrame(
   dart_frame_deadline_ = frame_target_time.ToEpochDelta();
   uint64_t frame_number = frame_timings_recorder_->GetFrameNumber();
   delegate_.OnAnimatorBeginFrame(frame_target_time, frame_number);
+}
+
+void Animator::EndFrame() {
+  FML_CHECK(frame_timings_recorder_ != nullptr);
+  if (!layer_trees_tasks_.empty()) {
+    // The build is completed in OnAnimatorBeginFrame.
+    frame_timings_recorder_->RecordBuildEnd(fml::TimePoint::Now());
+
+    delegate_.OnAnimatorUpdateLatestFrameTargetTime(
+        frame_timings_recorder_->GetVsyncTargetTime());
+
+    // Commit the pending continuation.
+    PipelineProduceResult result =
+        producer_continuation_.Complete(std::make_unique<FrameItem>(
+            std::move(layer_trees_tasks_), std::move(frame_timings_recorder_)));
+
+    if (!result.success) {
+      FML_DLOG(INFO) << "Failed to commit to the pipeline";
```

Comment:
This falls through where it used to return early. 

Do we really want to continue below in this case?

Reply:
When I was developing this PR I had the same question, but I think the current way is keeping the old behavior.

Before the PR, although when `result.success` is false, `Animator::Render` returned early, it was all contained in `delegate_.OnAnimatorBeginFrame`, while the `if (!frame_scheduled_ && has_rendered_)` clause continued to be executed.

If this logic seems questionable, I can open an issue to remind further checking.